### PR TITLE
user REST application

### LINKED ISSUE #13: USER port REST application
Add USER REST application according to OpenAPI documentation.

### PR DESCRIPTION
**Pull request by GALJO**
In this pull request, I added new endpoints: GET /user CREATE /user DELETE /user PUT /user/change-passwd
I improved exceptions to be more relevant
I improved Open API docs accuracy and added development server to them

### CODE REVIEW

File: application/user/decors/authorization.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+from domain.jwt.jwt_port import JWTPort
+from domain.jwt.model.JWTDecodeCmd import JWTDecodeCmd
+
+
+def authorization(jwt: JWTPort):
+    def decorator(f):
+        @wraps(f)
+        def wrapper(*args, **kwargs):
+            try:
+                _jwt = request.headers["Authorization"]
```
Comment (gerardolenski) [inline application/user/decors/authorization.py]: Pls add logging here with authorization details

File: application/util/exception_utils.py [hunk 2]
```diff
@@ -0,0 +1,5 @@
+import logging
+
+
+def exception_handler(exc) -> tuple:
```
Comment (gerardolenski) [inline application/util/exception_utils.py]: Function name should be a verb. Maybe handle_exception?

File: domain/jwt/adapter/jwt/jwt_adapter.py [hunk 3]
```diff
@@ -41,10 +41,14 @@ def decode(self, cmd: JWTDecodeCmd) -> JWTPayload:
                 _payload["iat"],
                 _payload["user_id"]
             )
+            logging.debug(f"JWT decoded: {_payload}")
```
Comment (gerardolenski) [inline domain/jwt/adapter/jwt/jwt_adapter.py:44]: Do you verify exp?

File: infrastructure/postgres/model/UserEntity.py [hunk 4]
```diff
@@ -7,7 +7,7 @@
 
 
 class UserEntity(_base_):
-    __tablename__ = "user"
+    __tablename__ = "User"
     id = Column(UUID(as_uuid=True), primary_key=True, default=uuid4())
     username = Column(String(16), unique=True, nullable=False)
```
Comment (gerardolenski) [inline infrastructure/postgres/model/UserEntity.py]: Why username is so short? Only 16 chars?

File: domain/jwt/adapter/jwt/jwt_adapter.py (hunk 3 above)
Comment (GALJO) [reply domain/jwt/adapter/jwt/jwt_adapter.py:44]: It gets verified in `jwt.decode()`

File: application/user/decors/authorization.py [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+from domain.jwt.model.JWTDecodeCmd import JWTDecodeCmd
+
+
+def authorization(jwt: JWTPort):
+    def decorator(f):
+        @wraps(f)
+        def wrapper(*args, **kwargs):
+            try:
+                header = request.headers["Authorization"]
+                logging.debug(f"Authorizing user: header={header}")
```
Comment (gerardolenski) [inline application/user/decors/authorization.py]: Rather authorizing request, not user

Comment (gerardolenski) [inline application/user/decors/authorization.py]: I hope you do not logging full token anywhere?

File: application/user/user_rest_adapter.py [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+
+
+@_app_.route("/user/change-passwd", methods=["PUT"])
+@authorization(jwt)
+@headers_check({"Accept": "text/plain", "Content-Type": "application/json"})
+def change_password(user_id):
+    try:
+        logging.info(f"Processing change user password request: user_id={user_id}")
+        body = request.get_json()
+        body = UserChangePasswordRequest(body["currentPassword"], body["newPassword"])
```
Comment (gerardolenski) [inline application/user/user_rest_adapter.py:81]: Maybe you could add constructor method which inits UserChangePasswordRequest  directly from http request?

File: application/user/decors/authorization.py (hunk 5 above)
Comment (GALJO) [reply application/user/decors/authorization.py]: I do, you said it is public and can be logged

File: application/user/user_rest_adapter.py (hunk 6 above)
Comment (GALJO) [reply application/user/user_rest_adapter.py:81]: I'm using it in [tests](https://github.com/Iron-Hills-Dev/huell/blob/13-user-port-rest-application/tests/application/user/test_change_password_it.py)

**Review by gerardolenski — APPROVED** [review]

**Pull request merged**